From dab667099bae3f5c19b236ef71005b288886d56d Mon Sep 17 00:00:00 2001
From: Dmitry Dementyev <dementyev@google.com>
Date: Mon, 9 Nov 2020 16:44:08 -0800
Subject: [PATCH 17/61] [Q_asb_2021-01] Protect
 GrantCredentialsPermissionActivity against overlay.

Bug: 169763814
Test: manual
Change-Id: I15dd22791fcc61ef02b06ad51d9e4409d11c0181
(cherry picked from commit f45dcfe1f5ed74dcec15799f2abd9e5ecbbed4b7)
---
 .../android/accounts/GrantCredentialsPermissionActivity.java   | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/java/android/accounts/GrantCredentialsPermissionActivity.java b/core/java/android/accounts/GrantCredentialsPermissionActivity.java
index af74b036a796..32b61b5ed8cc 100644
--- a/core/java/android/accounts/GrantCredentialsPermissionActivity.java
+++ b/core/java/android/accounts/GrantCredentialsPermissionActivity.java
@@ -47,6 +47,9 @@ public class GrantCredentialsPermissionActivity extends Activity implements View
 
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        getWindow().addSystemFlags(
+                android.view.WindowManager.LayoutParams
+                        .SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS);
         setContentView(R.layout.grant_credentials_permission);
         setTitle(R.string.grant_permissions_header_text);
 
-- 
2.34.1

